src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.exception.NoRequiredAnnotationException;
+import core.basesyntax.factory.BetDaoFactory;
+import core.basesyntax.factory.UserDaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                if (field.getType().equals(BetDao.class)) {
+                    BetDao betDao = BetDaoFactory.getBetDao();
+                    if (betDao.getClass().getAnnotation(Dao.class) == null) {
+                        throw new NoRequiredAnnotationException(
+                                "The BetDao class has no @Dao annotation");
+                    }
+                    field.setAccessible(true);
+                    field.set(instance, betDao);
+                }
+                if (field.getType().equals(UserDao.class)) {
+                    UserDao userDao = UserDaoFactory.getUserDao();
+                    if (userDao.getClass().getAnnotation(Dao.class) == null) {
+                        throw new NoRequiredAnnotationException(
+                                "The UserDao class has no @Dao annotation");
+                    }
+                    field.setAccessible(true);
+                    field.set(instance, userDao);
+                }
```

Comment:
let's refactor lines 20-37 to simplify the checks (line 19 is ok):

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception

Reply:
done